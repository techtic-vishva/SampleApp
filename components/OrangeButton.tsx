import React from 'react'
import { TouchableOpacity, StyleSheet, GestureResponderEvent, StyleProp, ViewStyle, ColorValue } from 'react-native'
import { darkArts, dichotomousHippopotamus, dichotomousHippopotamusDisabled } from '../constants/Colors'
import { Text } from './Themed'
import { Feather } from '@expo/vector-icons'

export type ButtonProps = {
	title: string
	icon?: React.ComponentProps<typeof Feather>['name']
	outterStyle?: StyleProp<ViewStyle>
	onPress?: (event: GestureResponderEvent) => void
	disabled?: boolean
	iconLeft?: boolean
	onLongPress?: (event: GestureResponderEvent) => void
	textColorOverride?: ColorValue
	iconColorOverride?: ColorValue
}

export default function (props: ButtonProps) {
	return (
		<TouchableOpacity
			disabled={props.disabled ?? false}
			style={[styles.base, props.outterStyle || {}, props.disabled ? styles.disabled : {}]}
			onPress={props.onPress}
			onLongPress={props.onLongPress}>
			{props.icon && props.iconLeft === true && (
				<Feather
					size={20}
					style={styles.icon}
					name={props.icon}
					color={props.iconColorOverride || props.textColorOverride}
				/>
			)}
			<Text style={{ color: props.textColorOverride ?? darkArts, fontSize: 18 }}>{props.title}</Text>
			{props.icon && props.iconLeft !== true && (
				<Feather
					size={20}
					style={[styles.icon]}
					color={props.iconColorOverride || props.textColorOverride}
					name={props.icon}
				/>
			)}
		</TouchableOpacity>
	)
}

const styles = StyleSheet.create({
	base: {
		display: 'flex',
		flexDirection: 'row',
		alignItems: 'center',
		justifyContent: 'center',
		backgroundColor: dichotomousHippopotamus,
		paddingVertical: 12,
		paddingHorizontal: 30,
		width: '50%',
		borderColor: dichotomousHippopotamus,
		borderRadius: 10,
		borderWidth: 1
	},
	icon: {
		paddingHorizontal: 10
	},
	disabled: {
		backgroundColor: dichotomousHippopotamusDisabled,
		borderColor: dichotomousHippopotamusDisabled
	}
})
