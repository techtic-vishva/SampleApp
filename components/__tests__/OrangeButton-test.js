import * as React from 'react'
import renderer from 'react-test-renderer'
import OrangeButton from '../OrangeButton'

it('render when passing the props', () => {
	const tree = renderer
		.create(
			<OrangeButton
				title="continue"
				icon="share"
				textColorOverride="#0b0805"
				outterStyle={{}}
				onLongPress={() => {
					console.log('onLogPressed')
				}}
				onPress={() => {
					console.log('onPress')
				}}
				iconLeft={false}
				disabled={false}
			/>
		)
		.toJSON()
	expect(tree).toMatchSnapshot()
})
