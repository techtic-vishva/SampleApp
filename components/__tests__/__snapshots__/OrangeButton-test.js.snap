// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render when passing the props 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ed832d",
      "borderColor": "#ed832d",
      "borderRadius": 10,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 30,
      "paddingVertical": 12,
      "width": "50%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "objektiv",
        },
        Object {
          "color": "#0b0805",
          "fontSize": 18,
        },
      ]
    }
  >
    continue
  </Text>
  <View
    color="#0b0805"
    name="share"
    size={20}
    style={
      Array [
        Object {
          "paddingHorizontal": 10,
        },
      ]
    }
  />
</View>
`;
